lib/helpers.js

```diff
@@ -25,8 +25,12 @@ function buildOnSettled(done) {
       return done(error, null);
     }
 
-    var settledErrors = filter(result, { state: 'error' });
-    var settledResults = filter(result, { state: 'success' });
+    var settledErrors = filter(result, function(elem) {
+      return elem.state === 'error';
+    });
+    var settledResults = filter(result, function(elem) {
```

Comment:
This anonymous function used as the 2nd parameter to `filter` can be a named function and moved to the top level (similar to the `getExtensions` function above)

You can probably name it `filterSuccess`